types/orchestration.d.ts

```diff
@@ -51,6 +51,15 @@ export interface OrchestrationTrigger extends FunctionTrigger {
     type: "orchestrationTrigger";
 }
 
+export type RegisteredOrchestration = (
+    input?: unknown,
+    instanceId?: string
+) => RegisteredOrchestrationTask;
+
+export interface RegisteredOrchestrationTask extends Task {
+    withRetry: (retryOptions: RetryOptions) => Task;
+}
```

Comment:
What happens if we do the following pseudocode:

```
var retryOptions = // something
var task = myRegisteredOrchestration();
var result1 = yield task;
var result2 = yield myRegisteredOrchestration().withRetry(retryOptions);
```

I think the final line, where we call `.withRetry` is weird edge case: the task has already completed (it's been yielded), and afterwards we've decided to add `.withRetry` options and yield again. From my understanding, we'll create a brand new task as a result and schedule it when we yield it. Is that correct? I don't think the behavior is immediately obvious, as I could also imagine that, since `task` already completed, the second `yield` will just fetch the pre-existing result from the history.

Reply:
Discussed offline. Changing this behavior to throw an error instead, since it is counterintuitive to spawn a new task from an already scheduled task. Implemented the change so that this throws an error. The same scenario now fails with this exception message:

```JSON
{
    "name": "testScenario",
    "instanceId": "9fd3813184b74a6c8bdd5efd278a510a",
    "runtimeStatus": "Failed",
    "input": "",
    "customStatus": null,
    "output": "Orchestrator function 'testScenario' failed: This activity has already been scheduled. You cannot call withRetry on an already scheduled task. Create a new task instead.",
    "createdTime": "2023-08-01T19:22:25Z",
    "lastUpdatedTime": "2023-08-01T19:23:16Z"
}
```

Thankfully, the `alreadyScheduled` flag we added a while back made this change quite straightforward :)